gst/tensor_transform/tensor_transform.c

```diff
@@ -1190,27 +1193,18 @@ gst_tensor_transform_transform_caps (GstBaseTransform * trans,
   gst_tensor_config_init (&in_config);
   gst_tensor_config_init (&out_config);
 
-  if (direction == GST_PAD_SINK) {
-    if (gst_tensor_transform_read_caps (caps, &in_config)) {
-      gst_tensor_transform_convert_dimension (filter, direction,
-          &in_config.info, &out_config.info);
-    }
-
-    /**
-     * supposed same framerate from input configuration
-     */
-    out_config.rate_n = in_config.rate_n;
-    out_config.rate_d = in_config.rate_d;
+  if (gst_tensor_transform_read_caps (caps, &in_config)) {
```

Comment:
Not safe when ```direction == GST_PAD_SRC```.
(needs ```s/in/out/``` for line 1196 if it's ```GST_PAD_SRC```)

Reply:
I removed the condition if (direction == GST_PAD_SINK), because gst_tensor_transform_convert_dimension() returns the result with pad direction.

Reply:
Ah.. got it. You are right!